### DISCUSSION TITLE
Search (<Ctrl><Shift>f search) dialog should mention that search format is special (AND, NOT, etc) !

### DISCUSSION THREAD

**Discussion by brunetton** [zim-desktop-wiki/zim-desktop-wiki#2565] [Ideas]
Hi

I'm using Zim Daily for years as my main notes taking app solution for work. Today I was configuring a Samba local mount, and I searched on my notes for `mount -t cifs` to see if I ever did something like this before. This gave me no result, so I started to take some time to read doc, and wanted to add this to my "Samba" Zim note. 

Here I realized that I had already did this someday before and already have this on my Samba note ! What a surprise. I thought that I found a bug in search system and started to do some testings (removing formatting, etc). It took me some time to realize that I could add double quotes around my search (`"mount -t cifs"`) to have the behavior I needed.

My suggestion is that *it should be more obvious* that this search is **silently** using a custom syntax **by default**. It's very tricky and very easy not to miss out this (as I probably done for years ! or less ? I've no idea since when this have been implemented)

I know there's an "Help" button, but as I was 100% sure that the search was simple, why should I need help just to make a grep ?

I think that this could be improved:
- by adding a mention to the help page, something saying "You can use 'AND', 'NOT', etc. For full documentation, click Help"
- by adding a checkbox to disable "advanced search" and making search behave like a good old fulltext search

What do you think about ?

Thanks !

**Comment by jaap-karssenberg** [comment]
The tool tip of the search box has exactly the help text you are looking
for.

To get literal text search the way of working is to quote the string. Not
unlike search in mayor search engines on which the syntax was inspired.

Only improvement I can think of is to add help text to the "no results"
view. Well push a patch for that

Regards,

Jaap


Op do 1 feb. 2024 13:06 schreef Bruno Duyé ***@***.***>:

**Reply by brunetton** [reply]
Thank you for your answer.

I looked to the tooltip and indeed, there is some information in it. I never seen this before ! I'm sorry but IMHO this isn't intuitive neither. I just asked to 2 friends of mine that are also using Zim as main note taking app and, like me, they never noticed that the search is not full-text. And I bet that the large majority of all users didn't noticed neither.

> To get literal text search the way of working is to quote the string

Yes I discovered that (the hard way). I'm not very happy to imagine having to wonder for all my searches (around 100 times a day): "Okay, is the word And in my search ? No ok, is the word And ? No, is there a minus ? Oh yes there is one; I've to quote my search then". This is quite time consuming and error-prone IMHO. 
And what about searches that contains double quotes ?

What do you think about this proposal below ? Do you think this would be a regression ? Or a useless option ?

![image](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/86322/1e0dd6a1-34ce-4fb2-b180-9fef2e9601a9)

My usage (maybe I don't represent the majority of users ?) is typically 100% full-text searching so I'd love to have this checkbox checked by default (remember the last value).

What do you think about ?

> Only improvement I can think of is to add help text to the "no results" view. Well push a patch for that

This is a good idea :+1:

**Comment by jaap-karssenberg** [comment]
To be honest I find the current syntax more intuitive. Based on how search
works in other applications as well.

If you ever two words, you expect matches with those two words but not
necessarily next to each other. I assume most users don't even think about
this. If you then want to filter down to exact matches you need to quote
the word into a single string. So some "advanced" syntax is actually
expected.

I admit first time you use a minus sign it may be unexpected, but when you
think about it, most other applications also require you to quote such
characters.

With the additional help text in the placeholder address feel it is good
for now.

Regards,

Jaap


Op do 1 feb. 2024 22:54 schreef Bruno Duyé ***@***.***>:

**Comment by brunetton** [comment]
Thank for your answer
